# Model Experiment: LightGBM

This notebook mirrors `model_experiment_XGBoost.ipynb`, using the same `WalmartBasePreprocessor` and `WalmartTabularFeatureEngineer` classes, so results are directly comparable between the two tree-based architectures.

LightGBM differs from XGBoost in how it grows trees (leaf-wise instead of level-wise) and in its native handling of missing values. Because of this, we also test an alternative feature set using the raw (NaN-preserved) markdown columns instead of the 0-filled + indicator version, to see if LightGBM's built-in missing-value handling performs better on its own

In [1]:
import dagshub
import mlflow
import lightgbm as lgb
import numpy as np
import pandas as pd
import optuna
import sys

sys.path.append("..")

from src.data.load_data import load_raw_data
from src.data.splits import calendar_aligned_split
from src.features.base import WalmartBasePreprocessor
from src.features.tabular import WalmartTabularFeatureEngineer

dagshub.init(repo_owner="LukaBatilashvili07", repo_name="walmart-sales-forecasting", mlflow=True)
mlflow.set_experiment("LightGBM_Training")
mlflow.lightgbm.autolog(log_models=False)

def wmae(y_true, y_pred, is_holiday):
    weights = np.where(is_holiday, 5, 1)
    return float(np.sum(weights * np.abs(y_true - y_pred)) / np.sum(weights))

Accessing as LukaBatilashvili07

Initialized MLflow to track repo "LukaBatilashvili07/walmart-sales-forecasting"

Repository LukaBatilashvili07/walmart-sales-forecasting initialized!

## 1. Data Loading and Train/Validation Split

Same split as XGBoost: the last 39 weeks of `train.csv` held out for validation, matching the length of the real `test.csv`

In [2]:
train_raw, test_raw, stores, features = load_raw_data("../data/raw")

train_part, valid_part = calendar_aligned_split(train_raw)

print(train_part["Date"].min(), "-", train_part["Date"].max())
print(valid_part["Date"].min(), "-", valid_part["Date"].max())

2010-02-05 00:00:00 - 2011-10-28 00:00:00
2011-11-04 00:00:00 - 2012-07-27 00:00:00


## 2. Run: LightGBM_Cleaning

Identical preprocessing to the XGBoost notebook, same `WalmartBasePreprocessor`, fit only on `stores`/`features`

In [3]:
with mlflow.start_run(run_name="LightGBM_Cleaning"):
    base_pre = WalmartBasePreprocessor()
    base_pre.fit(stores, features)

    train_base = base_pre.transform(train_part)
    valid_base = base_pre.transform(valid_part)

    mlflow.log_param("markdown_start_date", str(base_pre.MARKDOWN_START_DATE.date()))
    mlflow.log_metric("train_rows", len(train_base))
    mlflow.log_metric("valid_rows", len(valid_base))
    mlflow.log_metric("train_nan_total", int(train_base.isna().sum().sum()))
    mlflow.log_metric("valid_nan_total", int(valid_base.isna().sum().sum()))

    train_base.to_parquet("_cache_train_base_lgb.parquet")
    valid_base.to_parquet("_cache_valid_base_lgb.parquet")

🏃 View run LightGBM_Cleaning at: https://dagshub.com/LukaBatilashvili07/walmart-sales-forecasting.mlflow/#/experiments/3/runs/4adbfffcdef544ef850570f25d7ce517
🧪 View experiment at: https://dagshub.com/LukaBatilashvili07/walmart-sales-forecasting.mlflow/#/experiments/3


## 3. Run: LightGBM_Feature_Selection

Same feature set as XGBoost, plus one addition: the raw markdown columns are tested alongside the 0-filled versions. LightGBM can split on missing values natively

In [4]:
train_base = pd.read_parquet("_cache_train_base_lgb.parquet")
valid_base = pd.read_parquet("_cache_valid_base_lgb.parquet")

with mlflow.start_run(run_name="LightGBM_Feature_Selection"):
    tab_eng = WalmartTabularFeatureEngineer()
    tab_eng.fit(train_base[["Store", "Dept", "Date", "Weekly_Sales"]])

    train_full = tab_eng.transform_train(train_base)
    valid_full = tab_eng.transform_future(valid_base)

    feature_cols = [
        "Store", "Dept", "Size", "Temperature", "Fuel_Price",
        "CPI", "Unemployment",
        "Type_encoded",
        "Year", "Month", "WeekOfYear", "Week_sin", "Week_cos",
        "IsHoliday",
        "IsSuperBowl", "IsLaborDay", "IsThanksgiving", "IsChristmas",
        "MarkDown1_raw", "MarkDown2_raw", "MarkDown3_raw", "MarkDown4_raw", "MarkDown5_raw",
        "has_markdown_signal", "markdown_missing_count", "markdown_available_period",
        "Sales_lag_1", "Sales_lag_2", "Sales_lag_3", "Sales_lag_4", "Sales_lag_52",
        "Sales_diff_1",
        "Sales_roll_mean_4", "Sales_roll_std_4", "Sales_roll_mean_12",
        "Sales_seasonal_avg",
        "StoreDept_avg_sales", "Store_avg_sales", "Dept_avg_sales",
    ]

    X_train, y_train = train_full[feature_cols], train_full["Weekly_Sales"]
    X_valid, y_valid = valid_full[feature_cols], valid_full["Weekly_Sales"]

    w_train = np.where(train_full["IsHoliday"].values, 5, 1)

    baseline = lgb.LGBMRegressor(n_estimators=300, max_depth=6, random_state=42, n_jobs=-1)
    baseline.fit(X_train, y_train, sample_weight=w_train)

    importances = pd.Series(baseline.feature_importances_, index=feature_cols).sort_values(ascending=False)
    preds = baseline.predict(X_valid)
    baseline_wmae = wmae(y_valid.values, preds, valid_full["IsHoliday"].values)

    mlflow.log_param("n_features", len(feature_cols))
    mlflow.log_param("markdown_variant", "raw_nan")
    mlflow.log_metric("baseline_wmae", baseline_wmae)
    mlflow.log_metric("valid_wmae", baseline_wmae)
    importances.to_frame("importance").to_csv("_feature_importance_lgb.csv")
    mlflow.log_artifact("_feature_importance_lgb.csv")

    print(importances.head(15))
    print("Baseline WMAE:", baseline_wmae)

    train_full.to_parquet("_cache_train_full_lgb.parquet")
    valid_full.to_parquet("_cache_valid_full_lgb.parquet")

2026/07/24 09:23:23 WARNING mlflow.utils.autologging_utils: MLflow autologging encountered a warning: "/Users/lukabatilashvili/.pyenv/versions/3.9.16/lib/python3.9/site-packages/mlflow/types/utils.py:452: UserWarning: Hint: Inferred schema contains integer column(s). Integer columns in Python cannot represent missing values. If your input data contains missing values at inference time, it will be encoded as floats and will cause a schema enforcement error. The best way to avoid this problem is to infer the model schema based on a realistic data sample (training dataset) that includes missing values. Alternatively, you can declare integer columns as doubles (float64) whenever these columns may have missing values. See `Handling Integers With Missing Values <https://www.mlflow.org/docs/latest/models.html#handling-integers-with-missing-values>`_ for more details."


[LightGBM] [Info] Auto-choosing row-wise multi-threading, the overhead of testing was 0.007456 seconds.
You can set `force_row_wise=true` to remove the overhead.
And if memory is not enough, you can set `force_col_wise=true`.
[LightGBM] [Info] Total Bins 4261
[LightGBM] [Info] Number of data points in the train set: 267184, number of used features: 31
[LightGBM] [Info] Start training from score 16030.538151
[LightGBM] [Warning] No further splits with positive gain, best gain: -inf
[LightGBM] [Warning] No further splits with positive gain, best gain: -inf
[LightGBM] [Warning] No further splits with positive gain, best gain: -inf
[LightGBM] [Warning] No further splits with positive gain, best gain: -inf
[LightGBM] [Warning] No further splits with positive gain, best gain: -inf
[LightGBM] [Warning] No further splits with positive gain, best gain: -inf
[LightGBM] [Warning] No further splits with positive gain, best gain: -inf
[LightGBM] [Warning] No further splits with positive gain, best 

2026/07/24 09:23:30 WARNING mlflow.utils.autologging_utils: MLflow autologging encountered a warning: "/Users/lukabatilashvili/.pyenv/versions/3.9.16/lib/python3.9/site-packages/mlflow/types/utils.py:452: UserWarning: Hint: Inferred schema contains integer column(s). Integer columns in Python cannot represent missing values. If your input data contains missing values at inference time, it will be encoded as floats and will cause a schema enforcement error. The best way to avoid this problem is to infer the model schema based on a realistic data sample (training dataset) that includes missing values. Alternatively, you can declare integer columns as doubles (float64) whenever these columns may have missing values. See `Handling Integers With Missing Values <https://www.mlflow.org/docs/latest/models.html#handling-integers-with-missing-values>`_ for more details."


Sales_diff_1           1367
Sales_lag_1            1316
StoreDept_avg_sales    1059
Sales_seasonal_avg      585
Sales_roll_std_4        455
Unemployment            365
Sales_lag_2             332
Sales_roll_mean_4       319
WeekOfYear              299
Sales_roll_mean_12      299
IsThanksgiving          245
Dept                    240
Sales_lag_4             215
Temperature             200
Sales_lag_3             200
dtype: int32
Baseline WMAE: 5307.837095535916
🏃 View run LightGBM_Feature_Selection at: https://dagshub.com/LukaBatilashvili07/walmart-sales-forecasting.mlflow/#/experiments/3/runs/b46c818c4a3f46518be55ff1a78d586f
🧪 View experiment at: https://dagshub.com/LukaBatilashvili07/walmart-sales-forecasting.mlflow/#/experiments/3


## 4. Run: LightGBM_CV Walk-Forward Cross-Validation

Same walk-forward logic as the XGBoost notebook: three folds, each with a growing training window and a 13-week validation window right after it. This checks whether LightGBM's performance is stable across time periods, using the same fold boundaries as XGBoost for comparison

In [5]:
train_base = pd.read_parquet("_cache_train_base_lgb.parquet")

def make_walk_forward_folds(dates, n_folds=3, val_weeks=13):
    dates = sorted(dates)
    folds = []
    for i in range(n_folds, 0, -1):
        cutoff = len(dates) - i * val_weeks
        if cutoff <= 0:
            continue
        train_dates = dates[:cutoff]
        val_dates = dates[cutoff:cutoff + val_weeks]
        folds.append((train_dates, val_dates))
    return folds

unique_dates = train_base["Date"].unique()
folds = make_walk_forward_folds(unique_dates, n_folds=3, val_weeks=13)

with mlflow.start_run(run_name="LightGBM_CV"):
    fold_scores = []

    for i, (train_dates, val_dates) in enumerate(folds):
        fold_train_raw = train_base[train_base["Date"].isin(train_dates)]
        fold_val_raw = train_base[train_base["Date"].isin(val_dates)]

        fold_tab_eng = WalmartTabularFeatureEngineer()
        fold_tab_eng.fit(fold_train_raw[["Store", "Dept", "Date", "Weekly_Sales"]])

        fold_train_full = fold_tab_eng.transform_train(fold_train_raw)
        fold_val_full = fold_tab_eng.transform_future(fold_val_raw)

        X_tr = fold_train_full[feature_cols]
        y_tr = fold_train_full["Weekly_Sales"]
        X_val = fold_val_full[feature_cols]
        y_val = fold_val_full["Weekly_Sales"]

        model = lgb.LGBMRegressor(n_estimators=300, max_depth=6, random_state=42, n_jobs=-1)
        w_tr = np.where(fold_train_full["IsHoliday"].values, 5, 1)  # <-- დაამატე
        model.fit(X_tr, y_tr, sample_weight=w_tr)
        preds = model.predict(X_val)
        score = wmae(y_val.values, preds, fold_val_full["IsHoliday"].values)

        fold_scores.append(score)
        mlflow.log_metric(f"fold_{i}_wmae", score)
        print(f"Fold {i}: train up to {train_dates[-1]}, val {val_dates[0]}-{val_dates[-1]}, WMAE={score:.2f}")

    mlflow.log_metric("cv_wmae_mean", float(np.mean(fold_scores)))
    mlflow.log_metric("cv_wmae_std", float(np.std(fold_scores)))
    print("CV WMAE mean/std:", np.mean(fold_scores), np.std(fold_scores))

2026/07/24 09:23:39 WARNING mlflow.utils.autologging_utils: MLflow autologging encountered a warning: "/Users/lukabatilashvili/.pyenv/versions/3.9.16/lib/python3.9/site-packages/mlflow/types/utils.py:452: UserWarning: Hint: Inferred schema contains integer column(s). Integer columns in Python cannot represent missing values. If your input data contains missing values at inference time, it will be encoded as floats and will cause a schema enforcement error. The best way to avoid this problem is to infer the model schema based on a realistic data sample (training dataset) that includes missing values. Alternatively, you can declare integer columns as doubles (float64) whenever these columns may have missing values. See `Handling Integers With Missing Values <https://www.mlflow.org/docs/latest/models.html#handling-integers-with-missing-values>`_ for more details."


[LightGBM] [Info] Auto-choosing row-wise multi-threading, the overhead of testing was 0.004990 seconds.
You can set `force_row_wise=true` to remove the overhead.
And if memory is not enough, you can set `force_col_wise=true`.
[LightGBM] [Info] Total Bins 3814
[LightGBM] [Info] Number of data points in the train set: 152374, number of used features: 31
[LightGBM] [Info] Start training from score 16310.665653
[LightGBM] [Warning] No further splits with positive gain, best gain: -inf
[LightGBM] [Warning] No further splits with positive gain, best gain: -inf
[LightGBM] [Warning] No further splits with positive gain, best gain: -inf
[LightGBM] [Warning] No further splits with positive gain, best gain: -inf
[LightGBM] [Warning] No further splits with positive gain, best gain: -inf
[LightGBM] [Warning] No further splits with positive gain, best gain: -inf
[LightGBM] [Warning] No further splits with positive gain, best gain: -inf
[LightGBM] [Warning] No further splits with positive gain, best 

2026/07/24 09:23:43 WARNING mlflow.utils.autologging_utils: MLflow autologging encountered a warning: "/Users/lukabatilashvili/.pyenv/versions/3.9.16/lib/python3.9/site-packages/mlflow/types/utils.py:452: UserWarning: Hint: Inferred schema contains integer column(s). Integer columns in Python cannot represent missing values. If your input data contains missing values at inference time, it will be encoded as floats and will cause a schema enforcement error. The best way to avoid this problem is to infer the model schema based on a realistic data sample (training dataset) that includes missing values. Alternatively, you can declare integer columns as doubles (float64) whenever these columns may have missing values. See `Handling Integers With Missing Values <https://www.mlflow.org/docs/latest/models.html#handling-integers-with-missing-values>`_ for more details."


Fold 0: train up to 2011-01-28 00:00:00, val 2011-02-04 00:00:00-2011-04-29 00:00:00, WMAE=5177.87


2026/07/24 09:23:50 WARNING mlflow.utils.autologging_utils: MLflow autologging encountered a warning: "/Users/lukabatilashvili/.pyenv/versions/3.9.16/lib/python3.9/site-packages/mlflow/types/utils.py:452: UserWarning: Hint: Inferred schema contains integer column(s). Integer columns in Python cannot represent missing values. If your input data contains missing values at inference time, it will be encoded as floats and will cause a schema enforcement error. The best way to avoid this problem is to infer the model schema based on a realistic data sample (training dataset) that includes missing values. Alternatively, you can declare integer columns as doubles (float64) whenever these columns may have missing values. See `Handling Integers With Missing Values <https://www.mlflow.org/docs/latest/models.html#handling-integers-with-missing-values>`_ for more details."


[LightGBM] [Info] Auto-choosing row-wise multi-threading, the overhead of testing was 0.005336 seconds.
You can set `force_row_wise=true` to remove the overhead.
And if memory is not enough, you can set `force_col_wise=true`.
[LightGBM] [Info] Total Bins 4203
[LightGBM] [Info] Number of data points in the train set: 190674, number of used features: 31
[LightGBM] [Info] Start training from score 16171.109304
[LightGBM] [Warning] No further splits with positive gain, best gain: -inf
[LightGBM] [Warning] No further splits with positive gain, best gain: -inf
[LightGBM] [Warning] No further splits with positive gain, best gain: -inf
[LightGBM] [Warning] No further splits with positive gain, best gain: -inf
[LightGBM] [Warning] No further splits with positive gain, best gain: -inf
[LightGBM] [Warning] No further splits with positive gain, best gain: -inf
[LightGBM] [Warning] No further splits with positive gain, best gain: -inf
[LightGBM] [Warning] No further splits with positive gain, best 

2026/07/24 09:23:55 WARNING mlflow.utils.autologging_utils: MLflow autologging encountered a warning: "/Users/lukabatilashvili/.pyenv/versions/3.9.16/lib/python3.9/site-packages/mlflow/types/utils.py:452: UserWarning: Hint: Inferred schema contains integer column(s). Integer columns in Python cannot represent missing values. If your input data contains missing values at inference time, it will be encoded as floats and will cause a schema enforcement error. The best way to avoid this problem is to infer the model schema based on a realistic data sample (training dataset) that includes missing values. Alternatively, you can declare integer columns as doubles (float64) whenever these columns may have missing values. See `Handling Integers With Missing Values <https://www.mlflow.org/docs/latest/models.html#handling-integers-with-missing-values>`_ for more details."


Fold 1: train up to 2011-04-29 00:00:00, val 2011-05-06 00:00:00-2011-07-29 00:00:00, WMAE=4966.51


2026/07/24 09:24:02 WARNING mlflow.utils.autologging_utils: MLflow autologging encountered a warning: "/Users/lukabatilashvili/.pyenv/versions/3.9.16/lib/python3.9/site-packages/mlflow/types/utils.py:452: UserWarning: Hint: Inferred schema contains integer column(s). Integer columns in Python cannot represent missing values. If your input data contains missing values at inference time, it will be encoded as floats and will cause a schema enforcement error. The best way to avoid this problem is to infer the model schema based on a realistic data sample (training dataset) that includes missing values. Alternatively, you can declare integer columns as doubles (float64) whenever these columns may have missing values. See `Handling Integers With Missing Values <https://www.mlflow.org/docs/latest/models.html#handling-integers-with-missing-values>`_ for more details."


[LightGBM] [Info] Auto-choosing row-wise multi-threading, the overhead of testing was 0.005367 seconds.
You can set `force_row_wise=true` to remove the overhead.
And if memory is not enough, you can set `force_col_wise=true`.
[LightGBM] [Info] Total Bins 4232
[LightGBM] [Info] Number of data points in the train set: 228838, number of used features: 31
[LightGBM] [Info] Start training from score 16116.114367
[LightGBM] [Warning] No further splits with positive gain, best gain: -inf
[LightGBM] [Warning] No further splits with positive gain, best gain: -inf
[LightGBM] [Warning] No further splits with positive gain, best gain: -inf
[LightGBM] [Warning] No further splits with positive gain, best gain: -inf
[LightGBM] [Warning] No further splits with positive gain, best gain: -inf
[LightGBM] [Warning] No further splits with positive gain, best gain: -inf
[LightGBM] [Warning] No further splits with positive gain, best gain: -inf
[LightGBM] [Warning] No further splits with positive gain, best 

2026/07/24 09:24:06 WARNING mlflow.utils.autologging_utils: MLflow autologging encountered a warning: "/Users/lukabatilashvili/.pyenv/versions/3.9.16/lib/python3.9/site-packages/mlflow/types/utils.py:452: UserWarning: Hint: Inferred schema contains integer column(s). Integer columns in Python cannot represent missing values. If your input data contains missing values at inference time, it will be encoded as floats and will cause a schema enforcement error. The best way to avoid this problem is to infer the model schema based on a realistic data sample (training dataset) that includes missing values. Alternatively, you can declare integer columns as doubles (float64) whenever these columns may have missing values. See `Handling Integers With Missing Values <https://www.mlflow.org/docs/latest/models.html#handling-integers-with-missing-values>`_ for more details."


Fold 2: train up to 2011-07-29 00:00:00, val 2011-08-05 00:00:00-2011-10-28 00:00:00, WMAE=4829.07
CV WMAE mean/std: 4991.15094116958 143.45882080155187
🏃 View run LightGBM_CV at: https://dagshub.com/LukaBatilashvili07/walmart-sales-forecasting.mlflow/#/experiments/3/runs/1f3a32b39dfc45ed8bc326d5dc1012ad
🧪 View experiment at: https://dagshub.com/LukaBatilashvili07/walmart-sales-forecasting.mlflow/#/experiments/3


## 5. Run: LightGBM_HPO

Optuna searches LightGBM-specific parameters, most importantly num_leaves, which controls tree complexity in leaf growth. The objective is WMAE, same as XGBoost, for comparison. 

In [6]:
train_base = pd.read_parquet("_cache_train_base_lgb.parquet")

fold_data = []
for train_dates, val_dates in folds:
    fold_train_raw = train_base[train_base["Date"].isin(train_dates)]
    fold_val_raw = train_base[train_base["Date"].isin(val_dates)]

    fold_tab_eng = WalmartTabularFeatureEngineer()
    fold_tab_eng.fit(fold_train_raw[["Store", "Dept", "Date", "Weekly_Sales"]])

    fold_train_full = fold_tab_eng.transform_train(fold_train_raw)
    fold_val_full = fold_tab_eng.transform_future(fold_val_raw)

    fold_data.append({
        "X_tr": fold_train_full[feature_cols], "y_tr": fold_train_full["Weekly_Sales"],
        "X_val": fold_val_full[feature_cols], "y_val": fold_val_full["Weekly_Sales"],
        "w_tr": np.where(fold_train_full["IsHoliday"].values, 5, 1),
        "w_val": np.where(fold_val_full["IsHoliday"].values, 5, 1),
        "holiday_val": fold_val_full["IsHoliday"].values,
    })


def objective(trial):
    params = {
        "objective": "regression_l1",
        "n_estimators": trial.suggest_int("n_estimators", 200, 800, step=100),
        "num_leaves": trial.suggest_int("num_leaves", 15, 255),
        "max_depth": trial.suggest_int("max_depth", -1, 10),
        "learning_rate": trial.suggest_float("learning_rate", 0.01, 0.2, log=True),
        "subsample": trial.suggest_float("subsample", 0.6, 1.0),
        "subsample_freq": trial.suggest_int("subsample_freq", 1, 7),
        "colsample_bytree": trial.suggest_float("colsample_bytree", 0.6, 1.0),
        "min_child_samples": trial.suggest_int("min_child_samples", 5, 100),
        "reg_lambda": trial.suggest_float("reg_lambda", 0.1, 5.0, log=True),
        "random_state": 42,
        "n_jobs": -1,
        "verbosity": -1,
    }

    fold_scores = []
    with mlflow.start_run(run_name=f"LightGBM_HPO_trial_{trial.number}", nested=True):
        for fd in fold_data:
            model = lgb.LGBMRegressor(**params)
            model.fit(
                fd["X_tr"], fd["y_tr"],
                sample_weight=fd["w_tr"],
                eval_set=[(fd["X_val"], fd["y_val"])],
                eval_sample_weight=[fd["w_val"]],
                callbacks=[lgb.early_stopping(50, verbose=False)],
            )
            preds = model.predict(fd["X_val"])
            score = wmae(fd["y_val"].values, preds, fd["holiday_val"])
            fold_scores.append(score)

        mean_score = float(np.mean(fold_scores))
        mlflow.log_params(params)
        mlflow.log_metric("valid_wmae", mean_score)

    return mean_score


with mlflow.start_run(run_name="LightGBM_HPO"):
    study = optuna.create_study(direction="minimize")
    study.optimize(objective, n_trials=30)

    mlflow.log_params(study.best_params)
    mlflow.log_metric("valid_wmae", study.best_value)

best_params = study.best_params
print("Best parameters:", best_params)
print("Best WMAE:", study.best_value)

[I 2026-07-24 09:24:24,667] A new study created in memory with name: no-name-56c3d323-787d-4441-9f46-d16c96cdf94b
2026/07/24 09:24:27 WARNING mlflow.utils.autologging_utils: MLflow autologging encountered a warning: "/Users/lukabatilashvili/.pyenv/versions/3.9.16/lib/python3.9/site-packages/mlflow/types/utils.py:452: UserWarning: Hint: Inferred schema contains integer column(s). Integer columns in Python cannot represent missing values. If your input data contains missing values at inference time, it will be encoded as floats and will cause a schema enforcement error. The best way to avoid this problem is to infer the model schema based on a realistic data sample (training dataset) that includes missing values. Alternatively, you can declare integer columns as doubles (float64) whenever these columns may have missing values. See `Handling Integers With Missing Values <https://www.mlflow.org/docs/latest/models.html#handling-integers-with-missing-values>`_ for more details."
2026/07/24 0

🏃 View run LightGBM_HPO_trial_0 at: https://dagshub.com/LukaBatilashvili07/walmart-sales-forecasting.mlflow/#/experiments/3/runs/873bc45a9a6f47609387aba0c2ab9fa3
🧪 View experiment at: https://dagshub.com/LukaBatilashvili07/walmart-sales-forecasting.mlflow/#/experiments/3


[I 2026-07-24 09:25:45,744] Trial 0 finished with value: 4822.41691675231 and parameters: {'n_estimators': 500, 'num_leaves': 43, 'max_depth': 8, 'learning_rate': 0.02591428373949257, 'subsample': 0.6660437589572337, 'subsample_freq': 3, 'colsample_bytree': 0.9186137912117176, 'min_child_samples': 23, 'reg_lambda': 1.0564896164082636}. Best is trial 0 with value: 4822.41691675231.
2026/07/24 09:25:48 WARNING mlflow.utils.autologging_utils: MLflow autologging encountered a warning: "/Users/lukabatilashvili/.pyenv/versions/3.9.16/lib/python3.9/site-packages/mlflow/types/utils.py:452: UserWarning: Hint: Inferred schema contains integer column(s). Integer columns in Python cannot represent missing values. If your input data contains missing values at inference time, it will be encoded as floats and will cause a schema enforcement error. The best way to avoid this problem is to infer the model schema based on a realistic data sample (training dataset) that includes missing values. Alternati

🏃 View run LightGBM_HPO_trial_1 at: https://dagshub.com/LukaBatilashvili07/walmart-sales-forecasting.mlflow/#/experiments/3/runs/0ebf312e186c47b68436f3261b5af694
🧪 View experiment at: https://dagshub.com/LukaBatilashvili07/walmart-sales-forecasting.mlflow/#/experiments/3


[I 2026-07-24 09:26:54,281] Trial 1 finished with value: 5135.455244615736 and parameters: {'n_estimators': 500, 'num_leaves': 87, 'max_depth': 9, 'learning_rate': 0.035492562853095495, 'subsample': 0.8532005823584587, 'subsample_freq': 1, 'colsample_bytree': 0.8868745312450552, 'min_child_samples': 88, 'reg_lambda': 1.0288490665548944}. Best is trial 0 with value: 4822.41691675231.
2026/07/24 09:27:00 WARNING mlflow.utils.autologging_utils: MLflow autologging encountered a warning: "/Users/lukabatilashvili/.pyenv/versions/3.9.16/lib/python3.9/site-packages/mlflow/types/utils.py:452: UserWarning: Hint: Inferred schema contains integer column(s). Integer columns in Python cannot represent missing values. If your input data contains missing values at inference time, it will be encoded as floats and will cause a schema enforcement error. The best way to avoid this problem is to infer the model schema based on a realistic data sample (training dataset) that includes missing values. Alterna

🏃 View run LightGBM_HPO_trial_2 at: https://dagshub.com/LukaBatilashvili07/walmart-sales-forecasting.mlflow/#/experiments/3/runs/246213bfddbb4969a0aba7aabdeebf47
🧪 View experiment at: https://dagshub.com/LukaBatilashvili07/walmart-sales-forecasting.mlflow/#/experiments/3


[I 2026-07-24 09:28:29,894] Trial 2 finished with value: 4594.702868146261 and parameters: {'n_estimators': 400, 'num_leaves': 74, 'max_depth': -1, 'learning_rate': 0.028805428687182527, 'subsample': 0.6368933519920813, 'subsample_freq': 1, 'colsample_bytree': 0.7403250540649216, 'min_child_samples': 16, 'reg_lambda': 1.504927117529354}. Best is trial 2 with value: 4594.702868146261.
2026/07/24 09:28:36 WARNING mlflow.utils.autologging_utils: MLflow autologging encountered a warning: "/Users/lukabatilashvili/.pyenv/versions/3.9.16/lib/python3.9/site-packages/mlflow/types/utils.py:452: UserWarning: Hint: Inferred schema contains integer column(s). Integer columns in Python cannot represent missing values. If your input data contains missing values at inference time, it will be encoded as floats and will cause a schema enforcement error. The best way to avoid this problem is to infer the model schema based on a realistic data sample (training dataset) that includes missing values. Altern

🏃 View run LightGBM_HPO_trial_3 at: https://dagshub.com/LukaBatilashvili07/walmart-sales-forecasting.mlflow/#/experiments/3/runs/ddb51a49d8174a38897c7ceeb2c6640a
🧪 View experiment at: https://dagshub.com/LukaBatilashvili07/walmart-sales-forecasting.mlflow/#/experiments/3


[I 2026-07-24 09:29:31,695] Trial 3 finished with value: 4987.965354107414 and parameters: {'n_estimators': 200, 'num_leaves': 74, 'max_depth': 1, 'learning_rate': 0.1911915752761067, 'subsample': 0.7601514895481988, 'subsample_freq': 4, 'colsample_bytree': 0.9713528867786496, 'min_child_samples': 22, 'reg_lambda': 2.8380634555217203}. Best is trial 2 with value: 4594.702868146261.
2026/07/24 09:29:34 WARNING mlflow.utils.autologging_utils: MLflow autologging encountered a warning: "/Users/lukabatilashvili/.pyenv/versions/3.9.16/lib/python3.9/site-packages/mlflow/types/utils.py:452: UserWarning: Hint: Inferred schema contains integer column(s). Integer columns in Python cannot represent missing values. If your input data contains missing values at inference time, it will be encoded as floats and will cause a schema enforcement error. The best way to avoid this problem is to infer the model schema based on a realistic data sample (training dataset) that includes missing values. Alternat

🏃 View run LightGBM_HPO_trial_4 at: https://dagshub.com/LukaBatilashvili07/walmart-sales-forecasting.mlflow/#/experiments/3/runs/daa0242c94fa47ec936e968686db9d6a
🧪 View experiment at: https://dagshub.com/LukaBatilashvili07/walmart-sales-forecasting.mlflow/#/experiments/3


[I 2026-07-24 09:30:45,844] Trial 4 finished with value: 4689.194760720104 and parameters: {'n_estimators': 800, 'num_leaves': 247, 'max_depth': 9, 'learning_rate': 0.140724889182652, 'subsample': 0.7852247803520398, 'subsample_freq': 1, 'colsample_bytree': 0.782967504287295, 'min_child_samples': 91, 'reg_lambda': 1.1527596752121363}. Best is trial 2 with value: 4594.702868146261.
2026/07/24 09:30:49 WARNING mlflow.utils.autologging_utils: MLflow autologging encountered a warning: "/Users/lukabatilashvili/.pyenv/versions/3.9.16/lib/python3.9/site-packages/mlflow/types/utils.py:452: UserWarning: Hint: Inferred schema contains integer column(s). Integer columns in Python cannot represent missing values. If your input data contains missing values at inference time, it will be encoded as floats and will cause a schema enforcement error. The best way to avoid this problem is to infer the model schema based on a realistic data sample (training dataset) that includes missing values. Alternati

🏃 View run LightGBM_HPO_trial_5 at: https://dagshub.com/LukaBatilashvili07/walmart-sales-forecasting.mlflow/#/experiments/3/runs/a634ad8ac6c4457cadd36940199ba34c
🧪 View experiment at: https://dagshub.com/LukaBatilashvili07/walmart-sales-forecasting.mlflow/#/experiments/3


[I 2026-07-24 09:32:10,437] Trial 5 finished with value: 4606.5336066446625 and parameters: {'n_estimators': 800, 'num_leaves': 243, 'max_depth': 2, 'learning_rate': 0.10870021157270246, 'subsample': 0.6323331411063214, 'subsample_freq': 3, 'colsample_bytree': 0.8848656853050323, 'min_child_samples': 45, 'reg_lambda': 0.13716764996585404}. Best is trial 2 with value: 4594.702868146261.
2026/07/24 09:32:13 WARNING mlflow.utils.autologging_utils: MLflow autologging encountered a warning: "/Users/lukabatilashvili/.pyenv/versions/3.9.16/lib/python3.9/site-packages/mlflow/types/utils.py:452: UserWarning: Hint: Inferred schema contains integer column(s). Integer columns in Python cannot represent missing values. If your input data contains missing values at inference time, it will be encoded as floats and will cause a schema enforcement error. The best way to avoid this problem is to infer the model schema based on a realistic data sample (training dataset) that includes missing values. Alte

🏃 View run LightGBM_HPO_trial_6 at: https://dagshub.com/LukaBatilashvili07/walmart-sales-forecasting.mlflow/#/experiments/3/runs/93e0ee4490e5479d9c18552d15d1062c
🧪 View experiment at: https://dagshub.com/LukaBatilashvili07/walmart-sales-forecasting.mlflow/#/experiments/3


[I 2026-07-24 09:37:36,772] Trial 6 finished with value: 5778.940864492062 and parameters: {'n_estimators': 800, 'num_leaves': 223, 'max_depth': 1, 'learning_rate': 0.012757582324414683, 'subsample': 0.967639065493699, 'subsample_freq': 4, 'colsample_bytree': 0.8368151535859328, 'min_child_samples': 57, 'reg_lambda': 0.16337409946660614}. Best is trial 2 with value: 4594.702868146261.
2026/07/24 09:37:39 WARNING mlflow.utils.autologging_utils: MLflow autologging encountered a warning: "/Users/lukabatilashvili/.pyenv/versions/3.9.16/lib/python3.9/site-packages/mlflow/types/utils.py:452: UserWarning: Hint: Inferred schema contains integer column(s). Integer columns in Python cannot represent missing values. If your input data contains missing values at inference time, it will be encoded as floats and will cause a schema enforcement error. The best way to avoid this problem is to infer the model schema based on a realistic data sample (training dataset) that includes missing values. Alter

🏃 View run LightGBM_HPO_trial_7 at: https://dagshub.com/LukaBatilashvili07/walmart-sales-forecasting.mlflow/#/experiments/3/runs/fedc55e38108415b8d6c608374568819
🧪 View experiment at: https://dagshub.com/LukaBatilashvili07/walmart-sales-forecasting.mlflow/#/experiments/3


[I 2026-07-24 09:39:19,023] Trial 7 finished with value: 5251.083391212041 and parameters: {'n_estimators': 600, 'num_leaves': 98, 'max_depth': 4, 'learning_rate': 0.014533171693627988, 'subsample': 0.7509466274413085, 'subsample_freq': 5, 'colsample_bytree': 0.9350183452679653, 'min_child_samples': 11, 'reg_lambda': 0.2963568899262448}. Best is trial 2 with value: 4594.702868146261.
2026/07/24 09:39:25 WARNING mlflow.utils.autologging_utils: MLflow autologging encountered a warning: "/Users/lukabatilashvili/.pyenv/versions/3.9.16/lib/python3.9/site-packages/mlflow/types/utils.py:452: UserWarning: Hint: Inferred schema contains integer column(s). Integer columns in Python cannot represent missing values. If your input data contains missing values at inference time, it will be encoded as floats and will cause a schema enforcement error. The best way to avoid this problem is to infer the model schema based on a realistic data sample (training dataset) that includes missing values. Altern

🏃 View run LightGBM_HPO_trial_8 at: https://dagshub.com/LukaBatilashvili07/walmart-sales-forecasting.mlflow/#/experiments/3/runs/030de471a61946ebaae85a29af6af920
🧪 View experiment at: https://dagshub.com/LukaBatilashvili07/walmart-sales-forecasting.mlflow/#/experiments/3


[I 2026-07-24 09:41:30,499] Trial 8 finished with value: 6168.284424162118 and parameters: {'n_estimators': 500, 'num_leaves': 36, 'max_depth': 1, 'learning_rate': 0.01601594924946002, 'subsample': 0.9561366444855418, 'subsample_freq': 5, 'colsample_bytree': 0.8145454622773803, 'min_child_samples': 74, 'reg_lambda': 0.12868964430290103}. Best is trial 2 with value: 4594.702868146261.
2026/07/24 09:41:33 WARNING mlflow.utils.autologging_utils: MLflow autologging encountered a warning: "/Users/lukabatilashvili/.pyenv/versions/3.9.16/lib/python3.9/site-packages/mlflow/types/utils.py:452: UserWarning: Hint: Inferred schema contains integer column(s). Integer columns in Python cannot represent missing values. If your input data contains missing values at inference time, it will be encoded as floats and will cause a schema enforcement error. The best way to avoid this problem is to infer the model schema based on a realistic data sample (training dataset) that includes missing values. Altern

🏃 View run LightGBM_HPO_trial_9 at: https://dagshub.com/LukaBatilashvili07/walmart-sales-forecasting.mlflow/#/experiments/3/runs/0592e619e3564890af04e85e3e3b87f5
🧪 View experiment at: https://dagshub.com/LukaBatilashvili07/walmart-sales-forecasting.mlflow/#/experiments/3


[I 2026-07-24 09:42:22,521] Trial 9 finished with value: 4701.788700634114 and parameters: {'n_estimators': 300, 'num_leaves': 146, 'max_depth': 2, 'learning_rate': 0.08853305586667505, 'subsample': 0.6036462628413933, 'subsample_freq': 3, 'colsample_bytree': 0.8843339047579679, 'min_child_samples': 40, 'reg_lambda': 0.6669104941717242}. Best is trial 2 with value: 4594.702868146261.
2026/07/24 09:42:25 WARNING mlflow.utils.autologging_utils: MLflow autologging encountered a warning: "/Users/lukabatilashvili/.pyenv/versions/3.9.16/lib/python3.9/site-packages/mlflow/types/utils.py:452: UserWarning: Hint: Inferred schema contains integer column(s). Integer columns in Python cannot represent missing values. If your input data contains missing values at inference time, it will be encoded as floats and will cause a schema enforcement error. The best way to avoid this problem is to infer the model schema based on a realistic data sample (training dataset) that includes missing values. Altern

🏃 View run LightGBM_HPO_trial_10 at: https://dagshub.com/LukaBatilashvili07/walmart-sales-forecasting.mlflow/#/experiments/3/runs/b533d93a25444a8888dd2b702bfd8a8c
🧪 View experiment at: https://dagshub.com/LukaBatilashvili07/walmart-sales-forecasting.mlflow/#/experiments/3


[I 2026-07-24 09:43:30,923] Trial 10 finished with value: 4862.264147056772 and parameters: {'n_estimators': 300, 'num_leaves': 173, 'max_depth': 6, 'learning_rate': 0.055543418286561015, 'subsample': 0.877730484137896, 'subsample_freq': 7, 'colsample_bytree': 0.6422989171859085, 'min_child_samples': 67, 'reg_lambda': 4.099834462757734}. Best is trial 2 with value: 4594.702868146261.
2026/07/24 09:43:37 WARNING mlflow.utils.autologging_utils: MLflow autologging encountered a warning: "/Users/lukabatilashvili/.pyenv/versions/3.9.16/lib/python3.9/site-packages/mlflow/types/utils.py:452: UserWarning: Hint: Inferred schema contains integer column(s). Integer columns in Python cannot represent missing values. If your input data contains missing values at inference time, it will be encoded as floats and will cause a schema enforcement error. The best way to avoid this problem is to infer the model schema based on a realistic data sample (training dataset) that includes missing values. Altern

🏃 View run LightGBM_HPO_trial_11 at: https://dagshub.com/LukaBatilashvili07/walmart-sales-forecasting.mlflow/#/experiments/3/runs/1cd5e3ba78444662b15186528bb9f57b
🧪 View experiment at: https://dagshub.com/LukaBatilashvili07/walmart-sales-forecasting.mlflow/#/experiments/3


[I 2026-07-24 09:44:59,146] Trial 11 finished with value: 4238.529310116064 and parameters: {'n_estimators': 700, 'num_leaves': 177, 'max_depth': -1, 'learning_rate': 0.07046365860712553, 'subsample': 0.6520137049812676, 'subsample_freq': 2, 'colsample_bytree': 0.7082876312968046, 'min_child_samples': 43, 'reg_lambda': 0.36056712050864015}. Best is trial 11 with value: 4238.529310116064.
2026/07/24 09:45:05 WARNING mlflow.utils.autologging_utils: MLflow autologging encountered a warning: "/Users/lukabatilashvili/.pyenv/versions/3.9.16/lib/python3.9/site-packages/mlflow/types/utils.py:452: UserWarning: Hint: Inferred schema contains integer column(s). Integer columns in Python cannot represent missing values. If your input data contains missing values at inference time, it will be encoded as floats and will cause a schema enforcement error. The best way to avoid this problem is to infer the model schema based on a realistic data sample (training dataset) that includes missing values. Al

🏃 View run LightGBM_HPO_trial_12 at: https://dagshub.com/LukaBatilashvili07/walmart-sales-forecasting.mlflow/#/experiments/3/runs/0994ff2ba0514818a24708149b702da8
🧪 View experiment at: https://dagshub.com/LukaBatilashvili07/walmart-sales-forecasting.mlflow/#/experiments/3


[I 2026-07-24 09:46:26,599] Trial 12 finished with value: 4319.324566038008 and parameters: {'n_estimators': 600, 'num_leaves': 152, 'max_depth': -1, 'learning_rate': 0.04917494302941982, 'subsample': 0.6858304123060185, 'subsample_freq': 1, 'colsample_bytree': 0.7042925714426229, 'min_child_samples': 6, 'reg_lambda': 0.3490007613827072}. Best is trial 11 with value: 4238.529310116064.
2026/07/24 09:46:33 WARNING mlflow.utils.autologging_utils: MLflow autologging encountered a warning: "/Users/lukabatilashvili/.pyenv/versions/3.9.16/lib/python3.9/site-packages/mlflow/types/utils.py:452: UserWarning: Hint: Inferred schema contains integer column(s). Integer columns in Python cannot represent missing values. If your input data contains missing values at inference time, it will be encoded as floats and will cause a schema enforcement error. The best way to avoid this problem is to infer the model schema based on a realistic data sample (training dataset) that includes missing values. Alte

🏃 View run LightGBM_HPO_trial_13 at: https://dagshub.com/LukaBatilashvili07/walmart-sales-forecasting.mlflow/#/experiments/3/runs/718b22cc5bc54fad89191c7607aa601d
🧪 View experiment at: https://dagshub.com/LukaBatilashvili07/walmart-sales-forecasting.mlflow/#/experiments/3


[I 2026-07-24 09:58:30,117] Trial 13 finished with value: 4076.549114512624 and parameters: {'n_estimators': 700, 'num_leaves': 166, 'max_depth': -1, 'learning_rate': 0.05786499909185451, 'subsample': 0.6908041910258111, 'subsample_freq': 2, 'colsample_bytree': 0.666575519079867, 'min_child_samples': 36, 'reg_lambda': 0.39333443415976865}. Best is trial 13 with value: 4076.549114512624.
2026/07/24 09:58:33 WARNING mlflow.utils.autologging_utils: MLflow autologging encountered a warning: "/Users/lukabatilashvili/.pyenv/versions/3.9.16/lib/python3.9/site-packages/mlflow/types/utils.py:452: UserWarning: Hint: Inferred schema contains integer column(s). Integer columns in Python cannot represent missing values. If your input data contains missing values at inference time, it will be encoded as floats and will cause a schema enforcement error. The best way to avoid this problem is to infer the model schema based on a realistic data sample (training dataset) that includes missing values. Alt

🏃 View run LightGBM_HPO_trial_14 at: https://dagshub.com/LukaBatilashvili07/walmart-sales-forecasting.mlflow/#/experiments/3/runs/676cc1bcc76946b6a9659ad00b28a0ad
🧪 View experiment at: https://dagshub.com/LukaBatilashvili07/walmart-sales-forecasting.mlflow/#/experiments/3


[I 2026-07-24 10:00:01,077] Trial 14 finished with value: 3948.190728612866 and parameters: {'n_estimators': 700, 'num_leaves': 197, 'max_depth': -1, 'learning_rate': 0.06900906436408942, 'subsample': 0.7019571273799384, 'subsample_freq': 2, 'colsample_bytree': 0.6004202766172249, 'min_child_samples': 36, 'reg_lambda': 0.375490657506533}. Best is trial 14 with value: 3948.190728612866.
2026/07/24 10:00:04 WARNING mlflow.utils.autologging_utils: MLflow autologging encountered a warning: "/Users/lukabatilashvili/.pyenv/versions/3.9.16/lib/python3.9/site-packages/mlflow/types/utils.py:452: UserWarning: Hint: Inferred schema contains integer column(s). Integer columns in Python cannot represent missing values. If your input data contains missing values at inference time, it will be encoded as floats and will cause a schema enforcement error. The best way to avoid this problem is to infer the model schema based on a realistic data sample (training dataset) that includes missing values. Alte

🏃 View run LightGBM_HPO_trial_15 at: https://dagshub.com/LukaBatilashvili07/walmart-sales-forecasting.mlflow/#/experiments/3/runs/905c0bde63c345e4a0e75f182cd7eace
🧪 View experiment at: https://dagshub.com/LukaBatilashvili07/walmart-sales-forecasting.mlflow/#/experiments/3


[I 2026-07-24 10:01:14,131] Trial 15 finished with value: 4984.943768655004 and parameters: {'n_estimators': 700, 'num_leaves': 205, 'max_depth': 4, 'learning_rate': 0.04593327183804298, 'subsample': 0.7150282748681777, 'subsample_freq': 2, 'colsample_bytree': 0.6141517593909706, 'min_child_samples': 31, 'reg_lambda': 0.5646822039983646}. Best is trial 14 with value: 3948.190728612866.
2026/07/24 10:01:20 WARNING mlflow.utils.autologging_utils: MLflow autologging encountered a warning: "/Users/lukabatilashvili/.pyenv/versions/3.9.16/lib/python3.9/site-packages/mlflow/types/utils.py:452: UserWarning: Hint: Inferred schema contains integer column(s). Integer columns in Python cannot represent missing values. If your input data contains missing values at inference time, it will be encoded as floats and will cause a schema enforcement error. The best way to avoid this problem is to infer the model schema based on a realistic data sample (training dataset) that includes missing values. Alte

🏃 View run LightGBM_HPO_trial_16 at: https://dagshub.com/LukaBatilashvili07/walmart-sales-forecasting.mlflow/#/experiments/3/runs/7bc6f00a1d7948ac82af9e1ca4ccec4d
🧪 View experiment at: https://dagshub.com/LukaBatilashvili07/walmart-sales-forecasting.mlflow/#/experiments/3


[I 2026-07-24 10:02:26,194] Trial 16 finished with value: 4204.107623269443 and parameters: {'n_estimators': 700, 'num_leaves': 124, 'max_depth': 0, 'learning_rate': 0.0765799006822359, 'subsample': 0.8229385182818317, 'subsample_freq': 2, 'colsample_bytree': 0.6568171566529978, 'min_child_samples': 57, 'reg_lambda': 0.226355720074199}. Best is trial 14 with value: 3948.190728612866.
2026/07/24 10:02:29 WARNING mlflow.utils.autologging_utils: MLflow autologging encountered a warning: "/Users/lukabatilashvili/.pyenv/versions/3.9.16/lib/python3.9/site-packages/mlflow/types/utils.py:452: UserWarning: Hint: Inferred schema contains integer column(s). Integer columns in Python cannot represent missing values. If your input data contains missing values at inference time, it will be encoded as floats and will cause a schema enforcement error. The best way to avoid this problem is to infer the model schema based on a realistic data sample (training dataset) that includes missing values. Altern

🏃 View run LightGBM_HPO_trial_17 at: https://dagshub.com/LukaBatilashvili07/walmart-sales-forecasting.mlflow/#/experiments/3/runs/daf025e3ea79415bbec71238c0ac58af
🧪 View experiment at: https://dagshub.com/LukaBatilashvili07/walmart-sales-forecasting.mlflow/#/experiments/3


[I 2026-07-24 10:03:34,162] Trial 17 finished with value: 4618.862175480978 and parameters: {'n_estimators': 600, 'num_leaves': 200, 'max_depth': 3, 'learning_rate': 0.12989094233898443, 'subsample': 0.7204671620197174, 'subsample_freq': 7, 'colsample_bytree': 0.6033112928669901, 'min_child_samples': 32, 'reg_lambda': 0.5675271092171243}. Best is trial 14 with value: 3948.190728612866.
2026/07/24 10:03:40 WARNING mlflow.utils.autologging_utils: MLflow autologging encountered a warning: "/Users/lukabatilashvili/.pyenv/versions/3.9.16/lib/python3.9/site-packages/mlflow/types/utils.py:452: UserWarning: Hint: Inferred schema contains integer column(s). Integer columns in Python cannot represent missing values. If your input data contains missing values at inference time, it will be encoded as floats and will cause a schema enforcement error. The best way to avoid this problem is to infer the model schema based on a realistic data sample (training dataset) that includes missing values. Alte

🏃 View run LightGBM_HPO_trial_18 at: https://dagshub.com/LukaBatilashvili07/walmart-sales-forecasting.mlflow/#/experiments/3/runs/eb79a04a870849529bd58a5ef2a70513
🧪 View experiment at: https://dagshub.com/LukaBatilashvili07/walmart-sales-forecasting.mlflow/#/experiments/3


[I 2026-07-24 10:05:32,116] Trial 18 finished with value: 4181.66443596421 and parameters: {'n_estimators': 700, 'num_leaves': 129, 'max_depth': 0, 'learning_rate': 0.020657146077011513, 'subsample': 0.6940693374043936, 'subsample_freq': 2, 'colsample_bytree': 0.6806404080995718, 'min_child_samples': 55, 'reg_lambda': 0.21548081924806153}. Best is trial 14 with value: 3948.190728612866.
2026/07/24 10:05:35 WARNING mlflow.utils.autologging_utils: MLflow autologging encountered a warning: "/Users/lukabatilashvili/.pyenv/versions/3.9.16/lib/python3.9/site-packages/mlflow/types/utils.py:452: UserWarning: Hint: Inferred schema contains integer column(s). Integer columns in Python cannot represent missing values. If your input data contains missing values at inference time, it will be encoded as floats and will cause a schema enforcement error. The best way to avoid this problem is to infer the model schema based on a realistic data sample (training dataset) that includes missing values. Alt

🏃 View run LightGBM_HPO_trial_19 at: https://dagshub.com/LukaBatilashvili07/walmart-sales-forecasting.mlflow/#/experiments/3/runs/bf59a8fd4ac94e458ae745984a0116a3
🧪 View experiment at: https://dagshub.com/LukaBatilashvili07/walmart-sales-forecasting.mlflow/#/experiments/3


[I 2026-07-24 10:07:04,365] Trial 19 finished with value: 4241.940551865845 and parameters: {'n_estimators': 600, 'num_leaves': 181, 'max_depth': 0, 'learning_rate': 0.04055019681555051, 'subsample': 0.8118925205926457, 'subsample_freq': 4, 'colsample_bytree': 0.7419027058379823, 'min_child_samples': 33, 'reg_lambda': 0.439116988427818}. Best is trial 14 with value: 3948.190728612866.
2026/07/24 10:07:07 WARNING mlflow.utils.autologging_utils: MLflow autologging encountered a warning: "/Users/lukabatilashvili/.pyenv/versions/3.9.16/lib/python3.9/site-packages/mlflow/types/utils.py:452: UserWarning: Hint: Inferred schema contains integer column(s). Integer columns in Python cannot represent missing values. If your input data contains missing values at inference time, it will be encoded as floats and will cause a schema enforcement error. The best way to avoid this problem is to infer the model schema based on a realistic data sample (training dataset) that includes missing values. Alter

🏃 View run LightGBM_HPO_trial_20 at: https://dagshub.com/LukaBatilashvili07/walmart-sales-forecasting.mlflow/#/experiments/3/runs/4bbb6c1d267f4f089e2706ffea65ef1c
🧪 View experiment at: https://dagshub.com/LukaBatilashvili07/walmart-sales-forecasting.mlflow/#/experiments/3


[I 2026-07-24 10:08:23,069] Trial 20 finished with value: 4959.427588464296 and parameters: {'n_estimators': 800, 'num_leaves': 211, 'max_depth': 6, 'learning_rate': 0.06235323782830851, 'subsample': 0.7400215211191974, 'subsample_freq': 3, 'colsample_bytree': 0.6378092750526599, 'min_child_samples': 49, 'reg_lambda': 0.10158253315822904}. Best is trial 14 with value: 3948.190728612866.
2026/07/24 10:08:28 WARNING mlflow.utils.autologging_utils: MLflow autologging encountered a warning: "/Users/lukabatilashvili/.pyenv/versions/3.9.16/lib/python3.9/site-packages/mlflow/types/utils.py:452: UserWarning: Hint: Inferred schema contains integer column(s). Integer columns in Python cannot represent missing values. If your input data contains missing values at inference time, it will be encoded as floats and will cause a schema enforcement error. The best way to avoid this problem is to infer the model schema based on a realistic data sample (training dataset) that includes missing values. Alt

🏃 View run LightGBM_HPO_trial_21 at: https://dagshub.com/LukaBatilashvili07/walmart-sales-forecasting.mlflow/#/experiments/3/runs/0f51d5302b7e4e51ad11e8cd32afc074
🧪 View experiment at: https://dagshub.com/LukaBatilashvili07/walmart-sales-forecasting.mlflow/#/experiments/3


[I 2026-07-24 10:10:13,581] Trial 21 finished with value: 4372.395421892653 and parameters: {'n_estimators': 700, 'num_leaves': 124, 'max_depth': 0, 'learning_rate': 0.018660282110714358, 'subsample': 0.7029247622903173, 'subsample_freq': 2, 'colsample_bytree': 0.6899239758532006, 'min_child_samples': 57, 'reg_lambda': 0.18322972539286553}. Best is trial 14 with value: 3948.190728612866.
2026/07/24 10:10:16 WARNING mlflow.utils.autologging_utils: MLflow autologging encountered a warning: "/Users/lukabatilashvili/.pyenv/versions/3.9.16/lib/python3.9/site-packages/mlflow/types/utils.py:452: UserWarning: Hint: Inferred schema contains integer column(s). Integer columns in Python cannot represent missing values. If your input data contains missing values at inference time, it will be encoded as floats and will cause a schema enforcement error. The best way to avoid this problem is to infer the model schema based on a realistic data sample (training dataset) that includes missing values. Al

🏃 View run LightGBM_HPO_trial_22 at: https://dagshub.com/LukaBatilashvili07/walmart-sales-forecasting.mlflow/#/experiments/3/runs/0e74e37279ec4c36b5d77813d0ae9d79
🧪 View experiment at: https://dagshub.com/LukaBatilashvili07/walmart-sales-forecasting.mlflow/#/experiments/3


[I 2026-07-24 10:13:02,650] Trial 22 finished with value: 4246.085156929715 and parameters: {'n_estimators': 700, 'num_leaves': 146, 'max_depth': -1, 'learning_rate': 0.02280613249760028, 'subsample': 0.6810849286382421, 'subsample_freq': 2, 'colsample_bytree': 0.6756936994362193, 'min_child_samples': 65, 'reg_lambda': 0.2457419291700076}. Best is trial 14 with value: 3948.190728612866.
2026/07/24 10:13:05 WARNING mlflow.utils.autologging_utils: MLflow autologging encountered a warning: "/Users/lukabatilashvili/.pyenv/versions/3.9.16/lib/python3.9/site-packages/mlflow/types/utils.py:452: UserWarning: Hint: Inferred schema contains integer column(s). Integer columns in Python cannot represent missing values. If your input data contains missing values at inference time, it will be encoded as floats and will cause a schema enforcement error. The best way to avoid this problem is to infer the model schema based on a realistic data sample (training dataset) that includes missing values. Alt

🏃 View run LightGBM_HPO_trial_23 at: https://dagshub.com/LukaBatilashvili07/walmart-sales-forecasting.mlflow/#/experiments/3/runs/af4c13e7fbf747b59ca9552be86f9175
🧪 View experiment at: https://dagshub.com/LukaBatilashvili07/walmart-sales-forecasting.mlflow/#/experiments/3


[I 2026-07-24 10:14:24,431] Trial 23 finished with value: 4455.567790525146 and parameters: {'n_estimators': 700, 'num_leaves': 128, 'max_depth': 0, 'learning_rate': 0.03390800921770405, 'subsample': 0.6010031592090757, 'subsample_freq': 2, 'colsample_bytree': 0.7324729716317224, 'min_child_samples': 37, 'reg_lambda': 0.4622759202383681}. Best is trial 14 with value: 3948.190728612866.
2026/07/24 10:14:31 WARNING mlflow.utils.autologging_utils: MLflow autologging encountered a warning: "/Users/lukabatilashvili/.pyenv/versions/3.9.16/lib/python3.9/site-packages/mlflow/types/utils.py:452: UserWarning: Hint: Inferred schema contains integer column(s). Integer columns in Python cannot represent missing values. If your input data contains missing values at inference time, it will be encoded as floats and will cause a schema enforcement error. The best way to avoid this problem is to infer the model schema based on a realistic data sample (training dataset) that includes missing values. Alte

🏃 View run LightGBM_HPO_trial_24 at: https://dagshub.com/LukaBatilashvili07/walmart-sales-forecasting.mlflow/#/experiments/3/runs/f27fb2d1c9f5448ea0090c792fc71871
🧪 View experiment at: https://dagshub.com/LukaBatilashvili07/walmart-sales-forecasting.mlflow/#/experiments/3


[I 2026-07-24 10:15:52,468] Trial 24 finished with value: 4292.729463976343 and parameters: {'n_estimators': 600, 'num_leaves': 158, 'max_depth': 2, 'learning_rate': 0.09578984335730023, 'subsample': 0.7782704120479877, 'subsample_freq': 3, 'colsample_bytree': 0.6283928015464528, 'min_child_samples': 51, 'reg_lambda': 0.228743350617772}. Best is trial 14 with value: 3948.190728612866.
2026/07/24 10:15:59 WARNING mlflow.utils.autologging_utils: MLflow autologging encountered a warning: "/Users/lukabatilashvili/.pyenv/versions/3.9.16/lib/python3.9/site-packages/mlflow/types/utils.py:452: UserWarning: Hint: Inferred schema contains integer column(s). Integer columns in Python cannot represent missing values. If your input data contains missing values at inference time, it will be encoded as floats and will cause a schema enforcement error. The best way to avoid this problem is to infer the model schema based on a realistic data sample (training dataset) that includes missing values. Alter

🏃 View run LightGBM_HPO_trial_25 at: https://dagshub.com/LukaBatilashvili07/walmart-sales-forecasting.mlflow/#/experiments/3/runs/50de69a1754e48e48d265da1cc6078fb
🧪 View experiment at: https://dagshub.com/LukaBatilashvili07/walmart-sales-forecasting.mlflow/#/experiments/3


[I 2026-07-24 10:17:34,802] Trial 25 finished with value: 5809.795974911267 and parameters: {'n_estimators': 800, 'num_leaves': 188, 'max_depth': 1, 'learning_rate': 0.0112827600481058, 'subsample': 0.7122950326710942, 'subsample_freq': 1, 'colsample_bytree': 0.6639034593654043, 'min_child_samples': 76, 'reg_lambda': 0.7235126550340566}. Best is trial 14 with value: 3948.190728612866.
2026/07/24 10:17:39 WARNING mlflow.utils.autologging_utils: MLflow autologging encountered a warning: "/Users/lukabatilashvili/.pyenv/versions/3.9.16/lib/python3.9/site-packages/mlflow/types/utils.py:452: UserWarning: Hint: Inferred schema contains integer column(s). Integer columns in Python cannot represent missing values. If your input data contains missing values at inference time, it will be encoded as floats and will cause a schema enforcement error. The best way to avoid this problem is to infer the model schema based on a realistic data sample (training dataset) that includes missing values. Alter

🏃 View run LightGBM_HPO_trial_26 at: https://dagshub.com/LukaBatilashvili07/walmart-sales-forecasting.mlflow/#/experiments/3/runs/2c0d85c0dac848158eaf3ab439d9b2ca
🧪 View experiment at: https://dagshub.com/LukaBatilashvili07/walmart-sales-forecasting.mlflow/#/experiments/3


[I 2026-07-24 10:19:48,572] Trial 26 finished with value: 4374.229444332238 and parameters: {'n_estimators': 700, 'num_leaves': 227, 'max_depth': -1, 'learning_rate': 0.02027370000456329, 'subsample': 0.6709131930801174, 'subsample_freq': 2, 'colsample_bytree': 0.7693896446475578, 'min_child_samples': 21, 'reg_lambda': 0.29554503574299285}. Best is trial 14 with value: 3948.190728612866.
2026/07/24 10:19:51 WARNING mlflow.utils.autologging_utils: MLflow autologging encountered a warning: "/Users/lukabatilashvili/.pyenv/versions/3.9.16/lib/python3.9/site-packages/mlflow/types/utils.py:452: UserWarning: Hint: Inferred schema contains integer column(s). Integer columns in Python cannot represent missing values. If your input data contains missing values at inference time, it will be encoded as floats and will cause a schema enforcement error. The best way to avoid this problem is to infer the model schema based on a realistic data sample (training dataset) that includes missing values. Al

🏃 View run LightGBM_HPO_trial_27 at: https://dagshub.com/LukaBatilashvili07/walmart-sales-forecasting.mlflow/#/experiments/3/runs/f08845250d824902a97ccd44c745f033
🧪 View experiment at: https://dagshub.com/LukaBatilashvili07/walmart-sales-forecasting.mlflow/#/experiments/3


[I 2026-07-24 10:21:38,937] Trial 27 finished with value: 4399.5511642093 and parameters: {'n_estimators': 600, 'num_leaves': 115, 'max_depth': 0, 'learning_rate': 0.010369740700431887, 'subsample': 0.6371703253542962, 'subsample_freq': 3, 'colsample_bytree': 0.6035841009342622, 'min_child_samples': 27, 'reg_lambda': 0.4348186696232676}. Best is trial 14 with value: 3948.190728612866.
2026/07/24 10:21:41 WARNING mlflow.utils.autologging_utils: MLflow autologging encountered a warning: "/Users/lukabatilashvili/.pyenv/versions/3.9.16/lib/python3.9/site-packages/mlflow/types/utils.py:452: UserWarning: Hint: Inferred schema contains integer column(s). Integer columns in Python cannot represent missing values. If your input data contains missing values at inference time, it will be encoded as floats and will cause a schema enforcement error. The best way to avoid this problem is to infer the model schema based on a realistic data sample (training dataset) that includes missing values. Alter

🏃 View run LightGBM_HPO_trial_28 at: https://dagshub.com/LukaBatilashvili07/walmart-sales-forecasting.mlflow/#/experiments/3/runs/d4d278f9239342da85f93b6205238394
🧪 View experiment at: https://dagshub.com/LukaBatilashvili07/walmart-sales-forecasting.mlflow/#/experiments/3


[I 2026-07-24 10:22:46,937] Trial 28 finished with value: 4677.50998844389 and parameters: {'n_estimators': 500, 'num_leaves': 165, 'max_depth': 3, 'learning_rate': 0.1735820443659105, 'subsample': 0.734468596572708, 'subsample_freq': 1, 'colsample_bytree': 0.677133670855819, 'min_child_samples': 98, 'reg_lambda': 0.18961199345769084}. Best is trial 14 with value: 3948.190728612866.
2026/07/24 10:22:49 WARNING mlflow.utils.autologging_utils: MLflow autologging encountered a warning: "/Users/lukabatilashvili/.pyenv/versions/3.9.16/lib/python3.9/site-packages/mlflow/types/utils.py:452: UserWarning: Hint: Inferred schema contains integer column(s). Integer columns in Python cannot represent missing values. If your input data contains missing values at inference time, it will be encoded as floats and will cause a schema enforcement error. The best way to avoid this problem is to infer the model schema based on a realistic data sample (training dataset) that includes missing values. Alterna

🏃 View run LightGBM_HPO_trial_29 at: https://dagshub.com/LukaBatilashvili07/walmart-sales-forecasting.mlflow/#/experiments/3/runs/3109f60e19864f5f9d377da3f1118324
🧪 View experiment at: https://dagshub.com/LukaBatilashvili07/walmart-sales-forecasting.mlflow/#/experiments/3


[I 2026-07-24 10:24:47,110] Trial 29 finished with value: 4305.463322408828 and parameters: {'n_estimators': 800, 'num_leaves': 189, 'max_depth': -1, 'learning_rate': 0.026696690956989032, 'subsample': 0.6947875095290289, 'subsample_freq': 4, 'colsample_bytree': 0.7094179056397746, 'min_child_samples': 50, 'reg_lambda': 0.9509481388501311}. Best is trial 14 with value: 3948.190728612866.


🏃 View run LightGBM_HPO at: https://dagshub.com/LukaBatilashvili07/walmart-sales-forecasting.mlflow/#/experiments/3/runs/a95234c4087d425db7db4108790962ad
🧪 View experiment at: https://dagshub.com/LukaBatilashvili07/walmart-sales-forecasting.mlflow/#/experiments/3
Best parameters: {'n_estimators': 700, 'num_leaves': 197, 'max_depth': -1, 'learning_rate': 0.06900906436408942, 'subsample': 0.7019571273799384, 'subsample_freq': 2, 'colsample_bytree': 0.6004202766172249, 'min_child_samples': 36, 'reg_lambda': 0.375490657506533}
Best WMAE: 3948.190728612866


In [8]:
train_full = pd.read_parquet("_cache_train_full_lgb.parquet")
valid_full = pd.read_parquet("_cache_valid_full_lgb.parquet")

X_train, y_train = train_full[feature_cols], train_full["Weekly_Sales"]
X_valid, y_valid = valid_full[feature_cols], valid_full["Weekly_Sales"]
w_train = np.where(train_full["IsHoliday"].values, 5, 1)
holiday_valid = valid_full["IsHoliday"].values

## 6. Run: LightGBM_Best Final Model as a Pipeline

Same as the XGBoost pipeline: `WalmartLGBMPipeline` wraps fitted `base_preprocessor`, `tabular_engineer`, and trained LightGBM model in one object, so it can take a raw test set and return predictions.

In [9]:
class WalmartLGBMPipeline(mlflow.pyfunc.PythonModel):
    def __init__(self, base_preprocessor, tabular_engineer, model, feature_cols):
        self.base_preprocessor = base_preprocessor
        self.tabular_engineer = tabular_engineer
        self.model = model
        self.feature_cols = feature_cols

    def predict(self, context, model_input: pd.DataFrame):
        base = self.base_preprocessor.transform(model_input)
        full = self.tabular_engineer.transform_future(base)
        X = full[self.feature_cols]
        preds = self.model.predict(X)
        return np.maximum(preds, 0)


with mlflow.start_run(run_name="LightGBM_Best") as run:
    final_model = lgb.LGBMRegressor(
        **best_params,
        objective="regression_l1",
        random_state=42,
        n_jobs=-1,
        verbosity=-1,
    )
    final_model.fit(X_train, y_train, sample_weight=w_train)

    preds = final_model.predict(X_valid)
    final_wmae = wmae(y_valid.values, preds, holiday_valid)
    mlflow.log_params(best_params)
    mlflow.log_metric("final_valid_wmae", final_wmae)
    print("Final validation WMAE:", final_wmae)
    mlflow.log_metric("valid_wmae", final_wmae)

    pipeline = WalmartLGBMPipeline(
        base_preprocessor=base_pre,
        tabular_engineer=tab_eng,
        model=final_model,
        feature_cols=feature_cols,
    )

    mlflow.pyfunc.log_model(
        artifact_path="lightgbm_pipeline",
        python_model=pipeline,
        registered_model_name="Walmart_LightGBM_Pipeline",
    )

2026/07/24 10:28:25 WARNING mlflow.utils.autologging_utils: MLflow autologging encountered a warning: "/Users/lukabatilashvili/.pyenv/versions/3.9.16/lib/python3.9/site-packages/mlflow/types/utils.py:452: UserWarning: Hint: Inferred schema contains integer column(s). Integer columns in Python cannot represent missing values. If your input data contains missing values at inference time, it will be encoded as floats and will cause a schema enforcement error. The best way to avoid this problem is to infer the model schema based on a realistic data sample (training dataset) that includes missing values. Alternatively, you can declare integer columns as doubles (float64) whenever these columns may have missing values. See `Handling Integers With Missing Values <https://www.mlflow.org/docs/latest/models.html#handling-integers-with-missing-values>`_ for more details."
2026/07/24 10:29:19 WARNING mlflow.utils.autologging_utils: MLflow autologging encountered a warning: "/Users/lukabatilashvili

Final validation WMAE: 4449.903567727828


2026/07/24 10:29:22 WARNING mlflow.models.model: `artifact_path` is deprecated. Please use `name` instead.
2026/07/24 10:29:23 WARNING mlflow.pyfunc: Failed to infer model signature: Type hint <input: <class 'pandas.core.frame.DataFrame'>, output: None> cannot be used to infer model signature and input example is not provided, model signature cannot be inferred.
2026/07/24 10:29:28 WARNING mlflow.models.model: Model logged without a signature and input example. Please set `input_example` parameter when logging the model to auto infer the model signature.
Registered model 'Walmart_LightGBM_Pipeline' already exists. Creating a new version of this model...
2026/07/24 10:29:38 INFO mlflow.store.model_registry.abstract_store: Waiting up to 300 seconds for model version to finish creation. Model name: Walmart_LightGBM_Pipeline, version 5
Created version '5' of model 'Walmart_LightGBM_Pipeline'.


🏃 View run LightGBM_Best at: https://dagshub.com/LukaBatilashvili07/walmart-sales-forecasting.mlflow/#/experiments/3/runs/1254a1c598454eb7a1b846ac4c560763
🧪 View experiment at: https://dagshub.com/LukaBatilashvili07/walmart-sales-forecasting.mlflow/#/experiments/3


### Verifying the Pipeline Works on Raw Data

Same check as XGBoost: load the pipeline back by run ID and call `.predict()` directly on `test_raw`, the untouched test dataframe.

In [10]:
loaded = mlflow.pyfunc.load_model("models:/Walmart_LightGBM_Pipeline/5")
test_preds = loaded.predict(test_raw)
print(test_preds[:10])

2026/07/24 10:34:30 WARNING mlflow.utils.autologging_utils: MLflow autologging encountered a warning: "/Users/lukabatilashvili/.pyenv/versions/3.9.16/lib/python3.9/site-packages/mlflow/types/utils.py:452: UserWarning: Hint: Inferred schema contains integer column(s). Integer columns in Python cannot represent missing values. If your input data contains missing values at inference time, it will be encoded as floats and will cause a schema enforcement error. The best way to avoid this problem is to infer the model schema based on a realistic data sample (training dataset) that includes missing values. Alternatively, you can declare integer columns as doubles (float64) whenever these columns may have missing values. See `Handling Integers With Missing Values <https://www.mlflow.org/docs/latest/models.html#handling-integers-with-missing-values>`_ for more details."


[32448.53539308 18814.86645209 18899.05380824 19841.25268347
 20654.49479152 24129.56800453 27594.87289408 29455.91870023
 17984.18739296 18203.63347761]


## 7. Run: LightGBM_Final_Refit
Refit the best-params model on the ENTIRE train.csv (all 143 weeks), not just train_part (104 weeks)

In [11]:
with mlflow.start_run(run_name="LightGBM_Final_Refit") as run:
    full_base_pre = WalmartBasePreprocessor()
    full_base_pre.fit(stores, features)
    train_full_all = full_base_pre.transform(train_raw)

    full_tab_eng = WalmartTabularFeatureEngineer()
    full_tab_eng.fit(train_full_all[["Store", "Dept", "Date", "Weekly_Sales"]])
    train_full_all = full_tab_eng.transform_train(train_full_all)

    X_train_all = train_full_all[feature_cols]
    y_train_all = train_full_all["Weekly_Sales"]

    w_train_all = np.where(train_full_all["IsHoliday"].values, 5, 1)
    final_model_full = lgb.LGBMRegressor(
        **best_params,
        objective="regression_l1",
        random_state=42,
        n_jobs=-1,
        verbosity=-1,
    )
    final_model_full.fit(X_train_all, y_train_all, sample_weight=w_train_all)

    mlflow.log_params(best_params)
    mlflow.log_param("training_data", "full_train_143_weeks")
    mlflow.log_metric("train_rows", len(train_full_all))

    final_pipeline = WalmartLGBMPipeline(
        base_preprocessor=full_base_pre,
        tabular_engineer=full_tab_eng,
        model=final_model_full,
        feature_cols=feature_cols,
    )

    mlflow.pyfunc.log_model(
        artifact_path="lightgbm_pipeline_full",
        python_model=final_pipeline,
        registered_model_name="Walmart_LightGBM_Pipeline",
    )

    print("Final model trained on full train.csv:", len(train_full_all), "rows")

2026/07/24 10:34:59 WARNING mlflow.utils.autologging_utils: MLflow autologging encountered a warning: "/Users/lukabatilashvili/.pyenv/versions/3.9.16/lib/python3.9/site-packages/mlflow/types/utils.py:452: UserWarning: Hint: Inferred schema contains integer column(s). Integer columns in Python cannot represent missing values. If your input data contains missing values at inference time, it will be encoded as floats and will cause a schema enforcement error. The best way to avoid this problem is to infer the model schema based on a realistic data sample (training dataset) that includes missing values. Alternatively, you can declare integer columns as doubles (float64) whenever these columns may have missing values. See `Handling Integers With Missing Values <https://www.mlflow.org/docs/latest/models.html#handling-integers-with-missing-values>`_ for more details."
2026/07/24 10:36:06 WARNING mlflow.models.model: `artifact_path` is deprecated. Please use `name` instead.
2026/07/24 10:36:07

Final model trained on full train.csv: 421570 rows
🏃 View run LightGBM_Final_Refit at: https://dagshub.com/LukaBatilashvili07/walmart-sales-forecasting.mlflow/#/experiments/3/runs/54c1316d37f246d8ada249372985d3cb
🧪 View experiment at: https://dagshub.com/LukaBatilashvili07/walmart-sales-forecasting.mlflow/#/experiments/3
